server/server.go

```diff
@@ -510,17 +510,26 @@ func (s *Server) startMonitoring(secure bool) error {
 		hp           string
 		err          error
 		httpListener net.Listener
+		port   int
```

Comment:
On the port, conflicts, I haven't updated all the tests to use random selected ports. We'll continue to see intermittent port conflict errors until we do.